eclair-core/src/test/scala/fr/acinq/eclair/io/PeerSpec.scala

```diff
@@ -194,4 +195,20 @@ class PeerSpec extends TestkitBaseClass with StateTestsHelperMethods {
     assert(channelCreated.initialFeeratePerKw == peer.feeEstimator.getFeeratePerKw(peer.feeTargets.commitmentBlockTarget))
     assert(channelCreated.fundingTxFeeratePerKw.get == peer.feeEstimator.getFeeratePerKw(peer.feeTargets.fundingBlockTarget))
   }
+
+  test("use correct final script if option_static_remotekey is negotiated", Tag("static_remotekey")) { f =>
+    import f._
+
+    val probe = TestProbe()
+    connect(remoteNodeId, switchboard, peer, peerConnection, remoteInit = wire.Init(hex"2200")) // Bob supports option_static_remotekey
+    peer.stateData.channels.foreach { case (_, channelRef) =>
```

Comment:
This doesn't test anything as `peer.stateData.channels` is empty :)
I think you should test that when we send a message to trigger a channel open, the peer uses the right channel version when sending the channel command.

Reply:
I wanted to test that we both use the right `ChannelVersion` and we compute the final scriptPubkey with the localPaymentBasepoint (as done in [`createNewChannel()`](https://github.com/ACINQ/eclair/pull/1141/files#diff-51851274f6488b212bc8bfa2fa89797aR284)). I fixed the test in dd3326f